asset-transfer-ledger-queries/chaincode-go/asset_transfer_ledger_chaincode.go

```diff
@@ -260,63 +242,70 @@ func (t *SimpleChaincode) GetAssetsByRange(ctx contractapi.TransactionContextInt
 	return constructQueryResponseFromIterator(resultsIterator)
 }
 
-// TransferAssetBasedOnColor will transfer assets of a given color to a certain new owner.
-// Uses a GetStateByPartialCompositeKey (range query) against color~name 'index'.
+// TransferAssetByColor will transfer assets of a given color to a certain new owner.
+// Uses GetStateByPartialCompositeKey (range query) against color~name 'index'.
 // Committing peers will re-execute range queries to guarantee that result sets are stable
 // between endorsement time and commit time. The transaction is invalidated by the
 // committing peers if the result set has changed between endorsement time and commit time.
 // Therefore, range queries are a safe option for performing update transactions based on query results.
 // Example: GetStateByPartialCompositeKey/RangeQuery
-func (t *SimpleChaincode) TransferAssetBasedOnColor(ctx contractapi.TransactionContextInterface, color, newOwner string) error {
-	newOwner = strings.ToLower(newOwner)
-
-	// Query the color~name index by color
-	// This will execute a key range query on all keys starting with 'color'
+func (t *SimpleChaincode) TransferAssetByColor(ctx contractapi.TransactionContextInterface, color, newOwner string) error {
+	// Execute a key range query on all keys starting with 'color'
 	coloredAssetResultsIterator, err := ctx.GetStub().GetStateByPartialCompositeKey("color~name", []string{color})
```

Comment:
"color~name" index name is used in couple of places by now. Call for a const.

Reply:
Agreed

Reply:
Done